applications/nrf_desktop/src/modules/fast_pair.c

```diff
@@ -32,18 +33,94 @@ static inline size_t bond_cnt(uint8_t local_id)
 	return cnt;
 }
 
-static inline bool can_pair(uint8_t bt_local_id)
+static bool can_pair(uint8_t bt_local_id)
 {
 	return (bond_cnt(bt_local_id) < CONFIG_DESKTOP_FAST_PAIR_MAX_LOCAL_ID_BONDS);
 }
 
-static inline void update_fast_pair_ui_indication(void)
+static bool in_pairing_mode(uint8_t bt_local_id)
+{
+	/* Implementation assumes that given local identity advertises in pairing mode until
+	 * it has a bonded peer.
+	 */
+	return (bond_cnt(bt_local_id) == 0);
+}
+
+static void update_fast_pair_ui_indication(void)
 {
 	bool show_ui_pairing = can_pair(local_id);
 
 	bt_le_adv_prov_fast_pair_show_ui_pairing(show_ui_pairing);
-	LOG_INF("%s Fast Pair not discoverable advertising UI indication",
-		(show_ui_pairing) ? ("Show") : ("Hide"));
+
+	if (in_pairing_mode(local_id)) {
+		LOG_INF("Fast Pair discoverable advertising");
+	} else {
+		LOG_INF("%s Fast Pair not discoverable advertising UI indication",
+			(show_ui_pairing) ? ("Show") : ("Hide"));
+	}
+}
+
+static enum bt_security_err pairing_accept(struct bt_conn *conn,
+					   const struct bt_conn_pairing_feat *const feat)
+{
+	ARG_UNUSED(feat);
+
+	enum bt_security_err ret;
+	struct bt_conn_info info;
+	int err = bt_conn_get_info(conn, &info);
+
+	if (err) {
+		LOG_ERR("Cannot get conn info (err %d)", err);
+		return BT_SECURITY_ERR_UNSPECIFIED;
+	}
+
+	if (in_pairing_mode(info.id)) {
+		LOG_INF("Accept normal Bluetooth pairing");
+		ret = BT_SECURITY_ERR_SUCCESS;
+	} else {
+		LOG_WRN("Normal Bluetooth pairing not allowed outside of pairing mode");
+		ret = BT_SECURITY_ERR_PAIR_NOT_ALLOWED;
+	}
+
+	return ret;
+}
+
+static int register_app_auth_cbs(void)
+{
+	if (!IS_ENABLED(CONFIG_DESKTOP_FAST_PAIR_LIMIT_NORMAL_PAIRING)) {
+		return 0;
+	}
+
+	static const struct bt_conn_auth_cb conn_auth_callbacks = {
+		.pairing_accept = pairing_accept,
+	};
```

Comment:
Please add a comment to mention that this authentication structure is overlayed by the Fast Pair authentication callbacks in case of the Fast Pair pairing